src/ElasticSearchRepository.php

```diff
@@ -57,24 +58,31 @@ public function __construct(
     }
 
     /**
+     * @param Identifiable $data
      * @throws AssertionFailedException
      */
     public function save(Identifiable $data): void
     {
-        Assertion::true(
-            $data instanceof $this->class_type,
+        if (!class_exists($this->class_type)) {
+            throw new InvalidArgumentException(
+                "The class type provided ({$this->class_type}) does not exists."
+            );
+        }
+
+        Assertion::isInstanceOf(
+            $data,
+            $this->class_type,
             "Data object should be of type {$this->class_type}, as declared on the repository definition."
         );
```

Comment:
I've found out after some reading that this is what leaves phpstan confident that you know what you are doing. It adds an extra check, what do you think about it?

More info: https://github.com/phpstan/phpstan/issues/2808